support/build/Makefile.rules

```diff
@@ -82,18 +82,19 @@ endef
 #
 ################################################################################
 
+# Returns the fully-qualified path for a subfolder under the build output
+# directory
+sub_build_dir = $(addprefix $(BUILD_DIR)/,$(if $($(call uc,$(1))_BUILDTREE),\
+	$(1)/build,$(1)))
+
 # Creates a subfolder under the build output directory
 # mk_sub_build_dir $1:subfoldername
 define mk_sub_build_dir =
-$(if $(shell mkdir -p "$(BUILD_DIR)/$(notdir $(1))" && cd "$(BUILD_DIR)/$(notdir $(1))" >/dev/null && pwd),,\
-     $(error could not create directory "$(BUILD_DIR)/$(notdir $(1))"))
+$(eval _d := $(call sub_build_dir,$(1)))\
+$(if $(shell mkdir -p "$(_d)" && cd "$(_d)" >/dev/null && pwd),,\
```

Comment:
I tried to use $(MKDIR) (which already exists) for any invocation of `mkdir`. Maybe we should try to do the same, like here: [c00264def](https://github.com/unikraft/unikraft/pull/1034/commits/c00264def97a8c1d04a0211aa215949f0779a01d)